WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductRowViewModel.swift

```diff
@@ -67,9 +67,34 @@ final class ProductRowViewModel: ObservableObject, Identifiable {
         }
     }
 
-    /// Price label based on a product's price and quantity.
+    /// Provides a stock quantity label when applicable
     ///
-    private var priceLabel: String? {
+    var stockQuantityLabel: String {
+        createStockQuantityText()
+    }
+
+    /// Formatted price label for an individual product
+    ///
+    var priceLabel: String? {
+        guard let price = price else {
+            return nil
+        }
+        return currencyFormatter.formatAmount(price)
+    }
+
+    /// Formatted price label from multiplying product's price and quantity.
+    ///
+    var priceBeforeDiscountsLabel: String? {
+        guard let price = price else {
+            return nil
+        }
+        let productSubtotal = quantity * (currencyFormatter.convertToDecimal(price)?.decimalValue ?? Decimal.zero)
```

Comment:
This is something that felt odd as well: This bit it's copied from another method that performs the same logic, but then add discounts when needed, since I wanted to keep them separate. When testing what would happen if we pass a non-numeric-value, the currency formatter does return zero in any case, which I don't think is what we want in all cases, and feels a bit like counting on a hidden side effect in the currency formatter where we assume that a valid non-nil correct price will be passed into the method, and this will be correctly formatted, otherwise "it surely must be zero".

I opened a separate issue here: https://github.com/woocommerce/woocommerce-ios/issues/10875 so I can test further and improve it.

Reply:
I forgot to add, I added additional Unit Tests to the currency formatter for this method to show that both zero and non-numeric-values are converted: 1a48e053754e8868b7f127c63823b2f419f4f644 so it can help to change the behaviour later.